akka-stream/src/main/java-jdk9-only/akka/stream/javadsl/JavaFlowSupport.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http//www.lightbend.com/>
+ */
+
+package akka.stream.javadsl;
+
+import akka.NotUsed;
+import akka.japi.Pair;
+import akka.japi.Creator;
+import akka.stream.impl.JavaFlowAndRsConverters;
+
+/**
+ * For use only with `JDK 9+`.
+ * <p>
+ * Provides support for `java.util.concurrent.Flow.*` interfaces which mirror the Reactive Streams
+ * interfaces from `org.reactivestreams`. See <a href="http//www.reactive-streams.org/">reactive-streams.org</a>.
+ */
+public final class JavaFlowSupport {
+
+  private static final NotUsed NotUsed = akka.NotUsed.getInstance();
+
+  private JavaFlowSupport() {
+    throw new RuntimeException("No instances allowed!");
```

Comment:
can't this be enforced by making it `public final abstract class`?

Reply:
It is enforced by the constructor being private, but this is for crazy people who'd invoke it using reflection anyway